dgl_sparse/src/elemenwise_op.cc

```diff
@@ -22,6 +22,10 @@ c10::intrusive_ptr<SparseMatrix> SpSpAdd(
     const c10::intrusive_ptr<SparseMatrix>& A,
     const c10::intrusive_ptr<SparseMatrix>& B) {
   ElementwiseOpSanityCheck(A, B);
+  if (A->HasDiag() && B->HasDiag()) {
+    return SparseMatrix::FromDiagPointer(
+        A->DiagPtr(), A->value() + B->value(), A->shape());
+  }
```

Comment:
The current implementation is better.
For all functions, it could have multiple early return checks, so it is highly recommended to do:
```
if (...) {
  return ...
}
if (...) {
  return ...
}
blablabla
blablabla
blablabla
return ...
```